app/models/cart.rb

```diff
@@ -0,0 +1,30 @@
+class Cart
+  attr_reader :contents
+
+  def initialize(cart_info)
+    @contents = cart_info || {}
+  end
+
+  def total_count
+    contents.values.sum
+  end
+
+  def count_of(id)
+    contents[id.to_s] || 0
+  end
+
+  def add_freelancer(id)
+    contents[id.to_s] = (contents[id.to_s] || 0) + 1
+  end
+
+  def freelancer_ids
+    contents.keys
+  end
+
+  def total_price
+    #get the prices associated with each item, multiply by quantity, sum it up
```

Comment:
collect_freelancers.reduce do |sum, freelancer|
    sum + (freelancer.price * count_of(freelancer.id))
end

*** let's throw this is pry- but think this should work.

Reply:
thanks for the help with this method! 

we had to set the sum to 0 as an argument of reduce. otherwise, the sum was starting as a freelancer object.

I added a test and it's passing now.